cmd/state-svc/internal/resolver/analytics/analytics.go

```diff
@@ -169,8 +171,9 @@ func (r *Resolver) Events() chan<- deferred.EventData {
 	return r.events
 }
 
-func (r *Resolver) event(ev deferred.EventData) {
-	dimensions := r.customDimensions.toMap(ev.ProjectName, ev.Output, ev.UserID)
+func (r *Resolver) event(ev deferred.EventData, projectIDMap map[string]string) {
```

Comment:
It does not feel right that this function has such explicit awareness of the project ID. I would expect to see this logic live in the same place where we set the project owner/name dimension.

Reply:
I think you are correct, and the project ID should be resolved outside of this function.